test/e2e/quadlet_test.go

```diff
@@ -521,12 +536,13 @@ var _ = Describe("quadlet system generator", func() {
 			// Run quadlet to convert the file
 			session := podmanTest.Quadlet([]string{"--user", "-no-kmsg-log", generatedDir}, quadletDir)
 			session.WaitWithDefaultTimeout()
-			Expect(session).Should(Exit(0))
 
 			// Print any stderr output
 			errs := session.ErrorToString()
 			if errs != "" {
 				GinkgoWriter.Println("error:", session.ErrorToString())
+			} else {
+				Expect(session).Should(Exit(0))
```

Comment:
@edsantiago do you know how this test case is supposed to work.  I would like to add the exit code to the Entry functions below, but I have no idea how this is supposed to work.

Reply:
What @luap99 said; I would guess something like this (untested):
```diff
diff --git a/test/e2e/quadlet_test.go b/test/e2e/quadlet_test.go
index bfff92893..d46036f1a 100644
--- a/test/e2e/quadlet_test.go
+++ b/test/e2e/quadlet_test.go
@@ -526,7 +526,7 @@ BOGUS=foo
 	})
 
 	DescribeTable("Running quadlet test case",
-		func(fileName string) {
+		func(fileName string, expectedError string) {
 			testcase := loadQuadletTestcase(filepath.Join("quadlet", fileName))
 
 			// Write the tested file to the quadlet dir
@@ -536,15 +536,19 @@ BOGUS=foo
 			// Run quadlet to convert the file
 			session := podmanTest.Quadlet([]string{"--user", "-no-kmsg-log", generatedDir}, quadletDir)
 			session.WaitWithDefaultTimeout()
-
-			// Print any stderr output
 			errs := session.ErrorToString()
-			if errs != "" {
-				GinkgoWriter.Println("error:", session.ErrorToString())
+			if expectedError is nil {
+				Expect(err).ToNot(HaveOccurred())
+				// Print any stderr output
+				if errs != "" {
+					GinkgoWriter.Println("error:", session.ErrorToString())
+				}
 			} else {
-				Expect(session).Should(Exit(0))
+				Expect(err).To(HaveOccurred())
+				Expect(errs).To(ContainSubstring(expectedError))
 			}
 
+
 			testcase.check(generatedDir, session)
 		},
 		Entry("Basic container", "basic.container"),
```